src/test/java/io/cdap/delta/datastream/DatastreamTableRegistryTest.java

```diff
@@ -17,74 +16,66 @@
 import io.cdap.delta.api.assessment.TableList;
-import io.cdap.delta.api.assessment.TableRegistry;
 import io.cdap.delta.api.assessment.TableSummary;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
 
+import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-class DatastreamTableRegistryTest extends BaseIntegrationTestCase {
-
+public class DatastreamTableRegistryTest {
   @Test
-  public void testListDescribeTableNew() throws Exception {
-    testListDesribeTable(false);
-  }
+  public void testListTable() throws Exception {
+    DatastreamConfig datastreamConfig = Mockito.mock(DatastreamConfig.class);
+    Mockito.when(datastreamConfig.isUsingExistingStream()).thenReturn(false);
+    Mockito.when(datastreamConfig.getSid()).thenReturn("sid");
+    Mockito.when(datastreamConfig.getHost()).thenReturn("host");
+    Mockito.when(datastreamConfig.getUser()).thenReturn("user");
+    Mockito.when(datastreamConfig.getPassword()).thenReturn("password");
+    Mockito.when(datastreamConfig.getConnectivityMethod()).thenReturn("ip-allowlisting");
 
-  @Test
-  public void testListDescribeTableExisting() throws Exception {
-    testListDesribeTable(true);
-  }
+    DiscoverConnectionProfileResponse response = Mockito.mock(DiscoverConnectionProfileResponse.class);
+
+    OracleTable table = Mockito.mock(OracleTable.class);
+    Mockito.when(table.getTable()).thenReturn("table");
+    Mockito.when(table.getOracleColumnsCount()).thenReturn(2);
+
+    OracleSchema schema = Mockito.mock(OracleSchema.class);
+    Mockito.when(schema.getSchema()).thenReturn("schema");
+    Mockito.when(schema.getOracleTablesList()).thenReturn(Arrays.asList(table));
+
+    OracleRdbms oracleRdbms = Mockito.mock(OracleRdbms.class);
+    Mockito.when(response.getOracleRdbms()).thenReturn(oracleRdbms);
+    Mockito.when(oracleRdbms.getOracleSchemasList()).thenReturn(Arrays.asList(schema));
+
+    DatastreamClient datastreamClient = Mockito.mock(DatastreamClient.class);
+    Mockito.when(datastreamClient.discoverConnectionProfile(Mockito.any())).thenReturn(response);
+
+    DatastreamTableRegistry datastreamTableRegistry = new DatastreamTableRegistry(datastreamConfig, datastreamClient);
+    TableList tableList = datastreamTableRegistry.listTables();
 
-  private void testListDesribeTable(boolean usingExisting) throws Exception {
-    TableList tableList = null;
-    DatastreamDeltaSource deltaSource = createDeltaSource(usingExisting);
-    try (TableRegistry registry = deltaSource.createTableRegistry(null)) {
-      tableList = registry.listTables();
-    }
     assertNotNull(tableList);
     List<TableSummary> tables = tableList.getTables();
     assertNotNull(tables);
     assertFalse(tables.isEmpty());
```

Comment:
"Please add validations for tableList - schema/table name, table count" - validations are already added.